aten/src/TH/THAllocator.c

```diff
@@ -126,11 +137,50 @@ ptrdiff_t THMapAllocatorContext_size(THMapAllocatorContext *ctx)
 
 void THMapAllocatorContext_free(THMapAllocatorContext *ctx)
 {
-  if (ctx->filename != unknown_filename)
+  if (ctx->filename != unknown_filename) {
     THFree(ctx->filename);
+#ifdef _WIN32
+    THFree(ctx->eventname);
+#endif
+  }
   THFree(ctx);
 }
 
+#ifdef _WIN32
+typedef struct{
+  HANDLE event;
+  HANDLE handle;
+  void *data;
+} ReleaseContext;
+static DWORD WINAPI WaitForReleaseHandle(LPVOID lpParam) {
+  ReleaseContext *cxt = (ReleaseContext *)lpParam;
+
+  HANDLE event = cxt->event;
+  HANDLE handle = cxt->handle;
+  void *data = cxt->data;
+
+  while (TRUE) {
+    WaitForSingleObject(event, INFINITE);
+
+    THMapInfo *info = (THMapInfo*)(((char*)data) - TH_ALLOC_ALIGNMENT);
+    if (THAtomicGet(&info->refcount) == 0) {
```

Comment:
Why do you need to poll on this? The process that sets the even will do it only when it reaches zero, so you don't need to do it again in here, right?

Reply:
Will remove this.